src/main/kotlin/net/maizegenetics/phgv2/utils/VariantLoadingUtils.kt

```diff
@@ -0,0 +1,280 @@
+        .build()
+
+    val header = createGenericHeader(listOf(sampleName),altHeaderLines)
+    addSequenceDictionary(header, refGenomeSequence)
+    writer.writeHeader(header)
+    for(variant in variantContexts) {
+        writer.add(variant)
+    }
+
+    writer.close()
+}
+
+
+fun createGenericHeader(taxaNames: List<String>, altLines:Set<VCFHeaderLine>): VCFHeader {
+    val headerLines = createGenericHeaderLineSet() as MutableSet<VCFHeaderLine>
+    headerLines.addAll(altLines)
+    return VCFHeader(headerLines, taxaNames)
+}
+
+
+fun createGenericHeaderLineSet(): Set<VCFHeaderLine> {
+    val headerLines: MutableSet<VCFHeaderLine> = HashSet()
+    headerLines.add(VCFFormatHeaderLine("AD", 3, VCFHeaderLineType.Integer, "Allelic depths for the ref and alt alleles in the order listed"))
+    headerLines.add(
+        VCFFormatHeaderLine("DP", 1, VCFHeaderLineType.Integer, "Read Depth (only filtered reads used for calling)")
+    )
+    headerLines.add(VCFFormatHeaderLine("GQ", 1, VCFHeaderLineType.Integer, "Genotype Quality"))
+    headerLines.add(VCFFormatHeaderLine("GT", 1, VCFHeaderLineType.String, "Genotype"))
+    headerLines.add(
+        VCFFormatHeaderLine("PL", VCFHeaderLineCount.G, VCFHeaderLineType.Integer, "Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification")
+    )
+    headerLines.add(VCFInfoHeaderLine("DP", 1, VCFHeaderLineType.Integer, "Total Depth"))
+    headerLines.add(VCFInfoHeaderLine("NS", 1, VCFHeaderLineType.Integer, "Number of Samples With Data"))
+    headerLines.add(VCFInfoHeaderLine("AF", 3, VCFHeaderLineType.Integer, "Allele Frequency"))
+    headerLines.add(VCFInfoHeaderLine("END", 1, VCFHeaderLineType.Integer, "Stop position of the interval"))
+    headerLines.add(VCFInfoHeaderLine("ASM_Chr", 1, VCFHeaderLineType.String, "Assembly chromosome"))
+    headerLines.add(VCFInfoHeaderLine("ASM_Start", 1, VCFHeaderLineType.Integer, "Assembly start position"))
+    headerLines.add(VCFInfoHeaderLine("ASM_End", 1, VCFHeaderLineType.Integer, "Assembly end position"))
+    headerLines.add(VCFInfoHeaderLine("ASM_Strand", 1, VCFHeaderLineType.String, "Assembly strand"))
+    return headerLines
+}
+
+/**
+ * function to bgzip and create tabix index on a gvcf file
+ *
+ * This method uses the -f option on both bgzip and tabix to overwrite any existing files
+ * This handles the case where there already exists a bgzipped file, however
+ * it is still required to have the non-bzipped file present.
+ * @return the filename of the bgzipped gvcf
+ */
+fun bgzipAndIndexGVCFfile(gvcfFileName: String): String {
+
+    try {
+        // First bgzip the file
+
+        // bgzip the file - needed to create index
+        // use the -f option to overwrite any existing file
+        myLogger.info("bgzipping  file ${gvcfFileName}")
+        val gvcfGzippedFile = gvcfFileName + ".gz"
+        var builder = ProcessBuilder(
+            "bgzip", "-f", gvcfFileName)
```

Comment:
Not specific for this PR, but is bgzip/bcftools added to our conda package?

Reply:
Excellent question!  Probably not, and it should be.  I will look at changing that when I work on the SetupEnvironment branch.